site-configs.types.ts

```diff
@@ -0,0 +1,11 @@
+import type { SiteConfig as SiteConfigLibrary, SiteConfigPrivate, SiteConfigPublic } from "@comet/cli";
+
+export type SiteConfig = SiteConfigLibrary & {
+    contentScope: {
+        domain: string;
+        language: string;
+    }
+};
```

Comment:
Could we use module augmentation to narrow the type of contentScope? Or would this be a misuse of the feature?

Reply:
yes that might be even simpler. But not for contentScope, for the whole SiteConfig.

And to make it also work in site and admin you have to make sure that `../site-config.types.ts` is imported - although you don't need to import anything from there anymore. Maybe using `/// <reference types="..../site-configs.types" />` in vendors.d.ts (although that is not yet part of this PR)

Reply:
> I don't understand why this should make things simpler.

Having the application to re-define `SiteConfig` that is also defined as `SiteConfig` in the library is not ideal. That is the point where module agumentation is imho "simpler" - at least when using this type somewhere.

> Declaration merging does not allow to refine properties
true, that seems to be a show stopper I haven't thought of

> I'm even in doubt if ce98e4d does make things simpler because the type itself does not require defining the actual ContentSope type anymore

The application can define a ContentScope type if it needs it. You can also extract if from SiteConfig.